app/services/provider_interface/sort_application_choices.rb

```diff
@@ -1,24 +1,132 @@
 module ProviderInterface
   class SortApplicationChoices
-    def self.call(application_choices:, sort_by:)
-      application_choices.order(sort_order(sort_by))
+    def self.call(application_choices:)
+      for_task_view(application_choices).order(sort_order)
     end
 
-    def self.sort_order(sort_by)
-      return { updated_at: :desc } if sort_by != 'days_left_to_respond'
+    def self.for_task_view(application_choices)
+      application_choices.from <<~WITH_TASK_VIEW_GROUP.squish
+        (
+          SELECT a.*, c.recruitment_cycle_year,
+            CASE
+              WHEN #{deferred_offers_pending_reconfirmation} THEN 1
+              WHEN #{about_to_be_rejected_automatically} THEN 2
```

Comment:
No `3` ? This is present in the `TaskViewHelper#task_view_header` - is this redundant?

Reply:
No 3 is coming soon -- rejections without feedback (we don't have a feedback mechanism yet)